internal/server/authn/password_credentials_test.go

```diff
@@ -0,0 +1,228 @@
+package authn
+
+import (
+	"testing"
+
+	"golang.org/x/crypto/bcrypt"
+	"gorm.io/gorm"
+	"gotest.tools/v3/assert"
+
+	"github.com/infrahq/infra/internal/server/data"
+	"github.com/infrahq/infra/internal/server/models"
+)
+
+func TestPasswordCredentialAuthentication(t *testing.T) {
+	db := setupDB(t)
+
+	cases := map[string]map[string]interface{}{
```

Comment:
using a struct for setup/verify would be typesafe.

Reply:
I looked at this, but I liked keeping the setup/verify functions together for each case in the same map, so I'll leave it for now